### ISSUE TITLE
ModuleNotFoundError: No module named 'chardet'

### ISSUE THREAD

**Issue by johnhoman** [jupyterhub/repo2docker#1063]
### Bug description
Package dependency `chardet` not declared in setup file.

#### Expected behaviour
```bash
➜  repo2docker git:(master) mkvirtualenv -q repo2docker
(repo2docker) ➜  repo2docker git:(master) python -m pip install -e . -q
WARNING: You are using pip version 21.0.1; however, version 21.1.3 is available.
You should consider upgrading via the '/Users/jhoman/.virtualenvs/repo2docker/bin/python -m pip install --upgrade pip' command.
(repo2docker) ➜  repo2docker git:(master) python -c "import repo2docker.utils"
(repo2docker) ➜  repo2docker git:(master) echo $?
0
```
#### Actual behaviour
```sh
➜  repo2docker git:(master) mkvirtualenv -q repo2docker
(repo2docker) ➜  repo2docker git:(master) python -m pip install -e . -q
WARNING: You are using pip version 21.0.1; however, version 21.1.3 is available.
You should consider upgrading via the '/Users/jhoman/.virtualenvs/repo2docker/bin/python -m pip install --upgrade pip' command.
(repo2docker) ➜  repo2docker git:(master) python -c "import repo2docker.utils"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/jhoman/github.com/jupyterhub/repo2docker/repo2docker/__init__.py", line 6, in <module>
    from .app import Repo2Docker
  File "/Users/jhoman/github.com/jupyterhub/repo2docker/repo2docker/app.py", line 28, in <module>
    from .buildpacks import (
  File "/Users/jhoman/github.com/jupyterhub/repo2docker/repo2docker/buildpacks/__init__.py", line 2, in <module>
    from .python import PythonBuildPack
  File "/Users/jhoman/github.com/jupyterhub/repo2docker/repo2docker/buildpacks/python/__init__.py", line 4, in <module>
    from ..conda import CondaBuildPack
  File "/Users/jhoman/github.com/jupyterhub/repo2docker/repo2docker/buildpacks/conda/__init__.py", line 10, in <module>
    from ...utils import is_local_pip_requirement
  File "/Users/jhoman/github.com/jupyterhub/repo2docker/repo2docker/utils.py", line 6, in <module>
    import chardet
ModuleNotFoundError: No module named 'chardet'
(repo2docker) ➜  repo2docker git:(master) echo $?
1
```

### How to reproduce
1. Create a clean virtualenv
2. Clone the repo and install master
3. Try to import `repo2docker.utils` e.g. `python -c "import repo2docker.utils"` or just run `repo2docker --version`

### Your personal set up
Running on OSX with an ARM processor

```sh
(repo2docker) ➜  repo2docker git:(master) git describe --tags --dirty --always
2021.03.0-68-gdf65135
(repo2docker) ➜  repo2docker git:(master) git rev-parse --short HEAD
df65135
(repo2docker) ➜  repo2docker git:(master) repo2docker --version
Traceback (most recent call last):
  File "/Users/jhoman/.virtualenvs/repo2docker/bin/repo2docker", line 33, in <module>
    sys.exit(load_entry_point('jupyter-repo2docker', 'console_scripts', 'repo2docker')())
  File "/Users/jhoman/.virtualenvs/repo2docker/bin/repo2docker", line 25, in importlib_load_entry_point
    return next(matches).load()
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/importlib/metadata.py", line 77, in load
    module = import_module(match.group('module'))
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/jhoman/github.com/jupyterhub/repo2docker/repo2docker/__init__.py", line 6, in <module>
    from .app import Repo2Docker
  File "/Users/jhoman/github.com/jupyterhub/repo2docker/repo2docker/app.py", line 28, in <module>
    from .buildpacks import (
  File "/Users/jhoman/github.com/jupyterhub/repo2docker/repo2docker/buildpacks/__init__.py", line 2, in <module>
    from .python import PythonBuildPack
  File "/Users/jhoman/github.com/jupyterhub/repo2docker/repo2docker/buildpacks/python/__init__.py", line 4, in <module>
    from ..conda import CondaBuildPack
  File "/Users/jhoman/github.com/jupyterhub/repo2docker/repo2docker/buildpacks/conda/__init__.py", line 10, in <module>
    from ...utils import is_local_pip_requirement
  File "/Users/jhoman/github.com/jupyterhub/repo2docker/repo2docker/utils.py", line 6, in <module>
    import chardet
ModuleNotFoundError: No module named 'chardet'
```

It seems like the error isn't being pickup up regularly because the docker image that it runs in gets the dependency from another source https://github.com/jupyterhub/repo2docker/blob/df65135795a0ecf40dcaa0f15b62f6a85182d3e1/Dockerfile#L4
 
```sh
/ # apk add py-pip
(1/26) Installing py3-appdirs (1.4.4-r1)
(2/26) Installing py3-ordered-set (4.0.1-r0)
(3/26) Installing py3-parsing (2.4.7-r0)
(4/26) Installing py3-six (1.15.0-r0)
(5/26) Installing py3-packaging (20.4-r0)
(6/26) Installing py3-setuptools (47.0.0-r0)
**(7/26) Installing py3-chardet (3.0.4-r4)**
(8/26) Installing py3-idna (2.9-r0)
(9/26) Installing py3-certifi (2020.4.5.1-r0)
(10/26) Installing py3-urllib3 (1.25.9-r0)
(11/26) Installing py3-requests (2.23.0-r0)
(12/26) Installing py3-msgpack (1.0.0-r0)
(13/26) Installing py3-lockfile (0.12.2-r3)
(14/26) Installing py3-cachecontrol (0.12.6-r0)
(15/26) Installing py3-colorama (0.4.3-r0)
(16/26) Installing py3-distlib (0.3.0-r0)
(17/26) Installing py3-distro (1.5.0-r1)
(18/26) Installing py3-webencodings (0.5.1-r3)
(19/26) Installing py3-html5lib (1.0.1-r4)
(20/26) Installing py3-pytoml (0.1.21-r0)
(21/26) Installing py3-pep517 (0.8.2-r0)
(22/26) Installing py3-progress (1.5-r0)
(23/26) Installing py3-toml (0.10.1-r0)
(24/26) Installing py3-retrying (1.3.3-r0)
(25/26) Installing py3-contextlib2 (0.6.0-r0)
(26/26) Installing py3-pip (20.1.1-r0)
Executing busybox-1.31.1-r16.trigger
OK: 137 MiB in 57 packages
```

**Comment by betatim** [comment]
Do you want to create a PR that adds chardet to the list of dependencies in `setup.py`?

**Comment by manics** [comment]
Also bear in mind that repo2docker doesn't yet support building ARM images (https://github.com/jupyterhub/repo2docker/pull/1055). The missing dependency should still be fixed, but this may not be enough to get you a working repo2docker.

**Comment by johnhoman** [comment]
@betatim I'd be happy to #1064

**Comment by consideRatio** [comment]
Closed by #1064
